src/Services/QueuedJobService.php

```diff
@@ -1533,6 +1579,271 @@ private function addJobHandlersToLogger(LoggerInterface $logger, QueuedJob $job,
+        $remainingLimit = $jobRetryLimit;
+
+        foreach ($jobConfigs as $jobClass => $retryConfig) {
+            // We've reached the number of jobs we are allowed to retry in this health check run
+            if ($remainingLimit <= 0) {
+                break;
+            }
+
+            [
+                $maxRetryAttempts,
+            ] = $retryConfig;
+
+            $jobIDs = $autoRetryJobs
+                ->filter([
+                    'Implementation' => $jobClass,
+                    'RetryCount:LessThan' => $maxRetryAttempts,
+                ])
+                ->limit($remainingLimit)
+                ->column('ID');
+
+            // Add the found jobs to our list so we can retry these jobs
+            $retryJobIDs = array_merge($retryJobIDs, $jobIDs);
+
+            // Subtract found jobs from the remaining limit
+            $remainingLimit -= count($jobIDs);
+        }
+
+        // We couldn't find any jobs to retry
+        if (!$retryJobIDs) {
+            return;
+        }
+
+        $jobsToRetry = $autoRetryJobs->byIDs($retryJobIDs);
+
+        /** @var QueuedJobDescriptor $jobDescriptor */
+        foreach ($jobsToRetry as $jobDescriptor) {
+            $jobClass = $jobDescriptor->Implementation;
+            $retryConfig = array_key_exists($jobClass, $jobConfigs)
+                ? $jobConfigs[$jobClass]
+                : [];
+
+            // Couldn't find a retry config for this job (possible edge case in parallel processing)
+            if (!$retryConfig) {
+                continue;
+            }
+
+            [
+                $maxRetryAttempts,
+                $initialRetryDelay,
+                $retryFalloffMultiplier,
+                $retryFalloffMultiplierVariance,
+            ] = $retryConfig;
+
+            // Determine the config key that is currently active
+            $retryCount = max(0, $jobDescriptor->RetryCount);
+
+            // We have reached the maximum retry attempts for this job
+            if ($retryCount >= $maxRetryAttempts) {
+                continue;
+            }
+
+            /** @var QueuedJob $job */
+            $job = Injector::inst()->create($jobClass);
+
+            // Invalid implementation (possible edge case of invalid data)
+            if (!$job instanceof QueuedJob) {
+                continue;
+            }
+
+            if ($initialRetryDelay > 0) {
+                // Initial delay is configured - calculate the real delay value for this retry attempt
+
+                // We will use initial delay as our starting value
+                $delay = $initialRetryDelay;
+
+                // Calculate multiplier factoring in spread
+                $multiplier = $this->getJobRetryDelayMultiplier(
+                    $retryFalloffMultiplier,
+                    $retryFalloffMultiplierVariance
+                );
+
+                // Apply multiplier to total delay, factor in number of retries
+                $delay *= pow($multiplier, $retryCount);
```

Comment:
Why are we using `pow` here? Isn't it meant to just be directly multiplied (hence "multiplier" not "exponent")?

Reply:
I can confirm that the current implementation works as expected and the cluster breaking behaviour works well based on local testing. Will keep the current implementation. Will add an explicit note about this to the docs.